Documentation/concepts/observability/hubble-configuration.rst

```diff
@@ -70,3 +64,127 @@ only need to be provided when ``hubble.relay.tls.server.enabled`` is set to
+is used to schedule for certificates regeneration (regardless of their expiration date).
+
+.. code-block:: bash
+
+    --set hubble.tls.auto.enabled=true               # enable automatic TLS certificate generation
+    --set hubble.tls.auto.method=cronJob             # auto generate certificates using cronJob method
+    --set hubble.tls.auto.certValidityDuration=1095  # certificates validity duration in days (default 3 years)
+    --set hubble.tls.auto.schedule="0 0 1 */4 *"     # schedule for certificates re-generation (crontab syntax)
+
+Auto generated certificates via cert-manager
+--------------------------------------------
+
+This method rely on `cert-manager <https://cert-manager.io/>`__ to generated certificate.
+``cert-manager`` now become de-facto way to manage TLS on k8s, and compared to above methods,
+it has the following advantages:
+
+* No need to extra cronJob
+* Support multiple issuers: CA, Vault, Let's Encrypt, Google CAS,...
+  You can choose the issuer that fits your organization's requirements.
+* Manage certs via a CRD, which is more straightforward than inspecting the PEM file.
+* Auto-renew certificates
+
+**Installation steps**:
+
+1. First install `cert-manager <https://cert-manager.io/docs/installation/>`__ and setup `issuer <https://cert-manager.io/docs/configuration/>`_.
+   Please make sure that your issuer is be able to create certificates under the ``cilium.io`` domain name.
+2. Install/upgrade cilium with bellow configs:
+
+.. code-block:: bash
+
+    --set hubble.tls.auto.enabled=true               # enable automatic TLS certificate generation
+    --set hubble.tls.auto.method=certmanager         # auto generate certificates using cert-manager
+    --set hubble.tls.auto.certValidityDuration=1095  # certificates validity duration in days (default 3 years)
+    --set hubble.tls.auto.certManagerIssuerRef.group="cert-manager.io" # Reference to cert-manager's issuer
+    --set hubble.tls.auto.certManagerIssuerRef.kind="ClusterIssuer"
+    --set hubble.tls.auto.certManagerIssuerRef.name="ca-issuer"
+
+**Troubleshooting**:
+
+If you get the following error while install cilium (and cert-manager), it's because
+cert-manager ValidatingWebhook (which used to verify the Certificate CRD resources)
+is not available (blocked due to CNI is not available).
+
+.. code-block:: bash
+
+    Error: Internal error occurred: failed calling webhook "webhook.cert-manager.io": Post "https://cert-manager-webhook.cert-manager.svc:443/mutate?timeout=10s": dial tcp x.x.x.x:443: connect: connection refused
+
+There are several ways to overcome above issue, such as:
+
+* Option 1: disable validation on cilium installed namespace
+
+.. code-block:: bash
+
+    # We assum cilium installed in kube-system namespace
+    kubectl label namespace kube-system cert-manager.io/disable-validation=true
+
+    helm install cert-manager ...
+    kubectl apply -f issuer.yaml
+    helm install cilium ...
```

Comment:
For these ones, if you want to format them as a shell-session then you need to do:
```suggestion
.. code-block:: shell-session

    $ # We assume cilium installed in kube-system namespace
    $ kubectl label namespace kube-system cert-manager.io/disable-validation=true

    $ helm install cert-manager ...
    $ kubectl apply -f issuer.yaml
    $ helm install cilium ...
```

Reply:
Also, looking at these snippets, I see `issuer.yaml` mentioned but I'm not sure where that comes from. How does the user generate `issuer.yaml`?

Reply:
@joestringer I also add a note that where `issuer.yaml` is come from.